openedx/core/djangoapps/catalog/utils.py

```diff
@@ -130,21 +146,14 @@ def get_programs_with_type(types=None, include_hidden=True):
     if programs:
         program_types = {program_type['name']: program_type for program_type in get_program_types()}
         for program in programs:
-            # This limited type filtering is sufficient for current needs and
-            # helps us avoid additional complexity when caching program data.
-            # However, if the need for additional filtering of programs should
-            # arise, consider using the cache_programs management command to
-            # cache the filtered lists of UUIDs.
-            if types and program['type'] not in types:
-                continue
 
             if program['hidden'] and not include_hidden:
                 continue
 
             # deepcopy the program dict here so we are not adding
             # the type to the cached object
             program_with_type = copy.deepcopy(program)
-            program_with_type['type'] = program_types[program['type']]
+            program_with_type['type'] = program_types.get(program['type'])
```

Comment:
Leave this as it was. If the program type you need is missing, you want this to fail hard so that you know what the problem is right away.